File: github_cpp/21/15.cpp


#ifndef SELECTIONSORT_H_
#define SELECTIONSORT_H_

template<typename T>
class SelectionSort {
public:
	SelectionSort() {}
	virtual ~SelectionSort() {}

	static void sort(T data[], int n);

private:
	static void swap(T &a, T &b);
};

template<typename T>
void SelectionSort<T>::swap(T &a, T &b) {
	int temp = a;
	a = b;
	b = temp;
}

template<typename T>
void SelectionSort<T>::sort(T data[], int n) {
	for (int i = 0; i < n - 1; i ++) {
		int p = i;
		for (int j = 1; j < n; j++) {
			if (data[j] < data[p]) {
				p = j;
			}
		}

		if (p != i) {
			swap(data[i], data[p]);
		}
	}
}

#endif 

File: github_java_sort/9/15.java
public class Solution {
    
    public void sortIntegers(int[] A) {
        
        if (A == null) {
          return;
        }
        for (int i = 0; i < A.length; i++) {
          int minIndex = i;
          for (int j = i + 1; j < A.length; j++) {
            if (A[j] < A[minIndex]) {
              minIndex = j;
            }
          }
          if (minIndex != i) {
            swap(A, i, minIndex);
          }
        }
    }
    private void swap(int[] array, int i, int j) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
      }
}